fix(css): improve :defined pseudo-class example (#35371)
mdn/content: files/en-us/web/css/_colon_defined/index.md

### Hiding elements until they are defined

> In this demo, we define a basic custom element named `<custom-element>` and we use `:not(:defined)` and `:defined` selectors to style the element before and after it is defined. This is useful if you have a complex custom element that takes a while to load into the page — you might want to hide instances of the element until the definition is complete so that you don't end up with flashes of ugly unstyled elements on the page.

Reviewer: ```suggestion
In this demo, we define a basic custom element named `<custom-element>` and use the `:not(:defined)` and `:defined` selectors to style the element before and after it is defined. This is useful if you have a complex custom element that takes a while to load into the page — you might want to hide instances of the element until the definition is complete so that you don't end up with flashes of ugly unstyled elements on the page.
```

```html
<custom-element>
  <p>
    Loaded content: Lorem ipsum tel sed tellus eiusmod tellus. Aenean. Semper
    dolor sit nisi. Elit porttitor nisi sit vivamus.
  </p>
</custom-element>

<button id="btn">define the <code>&lt;custom-element&gt;</code></button>
```

> The above HTML code uses the custom element but the element hasn't been defined yet.

Reviewer: ```suggestion
The following HTML code uses the custom element but the element hasn't been defined yet.
```

...and put the line above the HTML.

Reviewer: Maybe also mention why the button is here? Something like "We also include a {{htmlelement("button")}} element that will define the custom element when clicked, allowing you to see the state of it before and after definition."

```css hidden
custom-element {
  display: block;
  border: 5px dashed grey;
  border-radius: 1rem;
  height: 100px;
  width: 400px;
  padding: 1rem;
  position: relative;
  user-select: none;
}

code {
  background: #ccc;
}

#btn {
  margin-top: 1rem;
  cursor: pointer;
}
```

> In the following code, we use `custom-element:not(:defined)` selector to select the element while it is not defined and use `custom-element:defined` selector to select the element after is defined. In the following CSS, we switch from grey to black color after the custom element is defined.

Reviewer: ```suggestion
In the following CSS, we use the `custom-element:not(:defined)` selector to select the element and color it grey while it is not defined and the `custom-element:defined` selector to select the element and color it black after it is defined.
```

```css
custom-element:not(:defined) {
  border-color: grey;
  color: grey;
}

custom-element:defined {
  background-color: wheat;
  border-color: black;
  color: black;
}

/* show loading message */
custom-element:not(:defined)::before {
  content: "Loading...";
  position: absolute;
  inset: 0 0 0 0;
  align-content: center;
  text-align: center;
  font-size: 2rem;
  background-color: white;
  border-radius: 1rem;
}

/* remove the loading message */
custom-element:defined::before {
  content: "";
}
```

> We have also used the [`::before`](/en-US/docs/Web/CSS/::before) pseudo-element to show the overlay loading message till the element is defined.

Reviewer: ```suggestion
We have also used the [`::before`](/en-US/docs/Web/CSS/::before) pseudo-element to show a "Loading..." overlay message until the element is defined. After definition, it is removed by setting the [`content`](/en-US/docs/Web/CSS/content) to an empty string.
```

> The following JavaScript has been used to define the custom element. In such a small demo the script executes instantly and it is not possible to see before definition styling. So instead of defining the custom element in a normal flow, we define the custom element with a click of the button.

Reviewer: ```suggestion
The following JavaScript has been used to define the custom element. As mentioned earlier, to allow you to see the state of the custom element before and after definition we run the {{domxref("CustomElementRegistry.define", "define()")}} method when the button is clicked.
```

```js
const btn = document.querySelector("#btn");

btn.addEventListener("click", () => {
  customElements.define("custom-element", class extends HTMLElement {});
  btn.remove();
});
```

> {{EmbedLiveSample("Hiding elements until they are defined", "100%", "230")}}

Reviewer: My only other thought — since this is a reference page, shouldn't you include "HTML", "CSS", "JavaScript", and "Result" subheadings in the demo section?
